kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -1287,4 +1336,149 @@ private static ConfigValue create(String origin, Object value) {
       return new AutoValue_KafkaRestConfig_ConfigValue(origin, value);
     }
   }
+
+  // this set contains configs that are a password type but not listed in AbstractConfig
+  private static final Set<String> passwordTypeConfigs =
+      ImmutableSet.of(
+          SaslConfigs.SASL_JAAS_CONFIG,
+          SchemaRegistryClientConfig.SCHEMA_REGISTRY_USER_INFO_CONFIG,
+          SchemaRegistryClientConfig.USER_INFO_CONFIG,
+          SchemaRegistryClientConfig.BEARER_AUTH_TOKEN_CONFIG,
+          SchemaRegistryClientConfig.BEARER_AUTH_CLIENT_SECRET);
+
+  /**
+   * This function relies on {@link AbstractConfig} object to detect whether a config is a {@link
+   * Password}. Therefore, it is important for the implementation to make sure that a config is
+   * registered correctly to not accidentally leaking sensitive information. If there is config that
+   * is sensitive but not registered correctly in its corresponding {@link AbstractConfig} object,
+   * please add that config to {@link KafkaRestConfig#passwordTypeConfigs} field above.
+   */
+  static <K, V> String mapToStringHideSensitiveConfigs(Map<K, V> map, AbstractConfig config) {
+    StringBuilder sb = new StringBuilder();
+    Set<Entry<K, V>> entries = map.entrySet();
+    sb.append('{');
+    Iterator<Entry<K, V>> it = entries.iterator();
+    while ((it.hasNext())) {
+      Entry<K, V> entry = it.next();
+      K key = entry.getKey();
+      V value = entry.getValue();
+      if (key instanceof String) {
+        if (passwordTypeConfigs.contains(key)) {
+          toStringKeyAndValue(sb, key, Password.HIDDEN, it.hasNext());
+        } else {
+          Type type = config.typeOf((String) key);
+          if (type == Type.PASSWORD) {
+            toStringKeyAndValue(sb, key, config.getPassword((String) key), it.hasNext());
```

Comment:
So we are redacting only if and only if it's a part of "passwordTypeConfigs" set, otherwise, we are showing it as it is, is it an intended logic?

Reply:
yeah, this is the logic, if our configurations are sane, then we should expect password type to be sensitive ones.

Reply:
Got it!

Reply:
Resolving the convo